### PR TITLE
Tracking pixels for content blocking fetch and lookup

### PR DESCRIPTION
**Pull request by studiosutara**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1208613456171888/1208801514911204/f
iOS PR: https://github.com/duckduckgo/iOS/pull/3597
macOS PR: 
What kind of version bump will this require?: Major/Minor/Patch

**Optional**:

Tech Design URL:
CC:

**Description**:

**Steps to test this PR**:
1.
1.

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/ContentBlocking/ContentBlockerRulesManager.swift [hunk 1]
```diff
@@ -377,6 +388,9 @@ public class ContentBlockerRulesManager: CompiledRuleListsSource {
                                                                                                 unprotectedSitesHash: nil))
             }
 
+            // if task is main tds task, extract time and iteration count from the result and fire the pixel
+            // can result be a struct? can we add fields to compliatiotask instead?
+            // can we sen dpixels from compialtion task?
```
Comment (bwaresiak) [inline Sources/BrowserServicesKit/ContentBlocking/ContentBlockerRulesManager.swift:393]: I think this is not an intended change :)

**Review by bwaresiak — COMMENTED** [review]
Great job, I've added one comment around a comment that probably was not intended there. :)

I think we also should be looking at how the event you've added are fired for `ContentBlockerRulesManagerInitialCompilationTests` can you add that please?

File: Sources/BrowserServicesKit/ContentBlocking/ContentBlockerRulesManager.swift (hunk 1 above)
Comment (studiosutara) [reply Sources/BrowserServicesKit/ContentBlocking/ContentBlockerRulesManager.swift]: Done

File: Tests/BrowserServicesKitTests/ContentBlocker/ContentBlockerRulesManagerInitialCompilationTests.swift [hunk 2]
```diff
@@ -240,20 +295,21 @@ final class ContentBlockerRulesManagerInitialCompilationTests: XCTestCase {
...
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.etag, oldEtag)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.name, mockRulesSource.ruleListName)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.trackerData, mockRulesSource.trackerData?.tds)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.identifier, oldIdentifier)
-        } else if expRecompiled.currentFulfillmentCount == 2 { // finished recompilation of rules due to changed tds
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.etag, updatedEtag)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.name, mockRulesSource.ruleListName)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.trackerData, mockRulesSource.trackerData?.tds)
-            XCTAssertEqual(mockLastCompiledRulesStore.rules.first?.identifier, newIdentifier)
+            if expRecompiled.currentFulfillmentCount == 1 { // finished compilation after cold start (using last compiled rules)
```
Comment (studiosutara) [inline Tests/BrowserServicesKitTests/ContentBlocker/ContentBlockerRulesManagerInitialCompilationTests.swift:299]: This was not needed for my work, but I noticed that code wasn’t getting called. I moved it up into the block. Please check that this is correct.

**Pull request merged**